pkg/image/apiserver/registry/imagestream/etcd/etcd.go

```diff
@@ -172,46 +175,73 @@ func (r *LayersREST) Get(ctx context.Context, name string, options *metav1.GetOp
 		Images:     make(map[string]imageapi.ImageBlobReferences),
 	}
 
+	missing := addImageStreamLayersFromCache(isl, is, r.index)
+	// if we are missing images, they may not have propogated to the cache. Wait a non-zero amount of time
+	// and try again
+	if len(missing) > 0 {
+		// 250ms is a reasonable propagation delay for a medium to large cluster
+		time.Sleep(250 * time.Millisecond)
+		missing = addImageStreamLayersFromCache(isl, is, r.index)
+		if len(missing) > 0 {
+			// TODO: return this in the API object as well
+			glog.V(2).Infof("Image stream %s/%s references %d images that could not be found: %v", is.Namespace, is.Name, len(missing), missing)
```

Comment:
under what circumstances (other than propagation delay) can this happen?  Someone manually deleting image objects?  pruning gone awry?

Reply:
Someone manually deleting images.  A rogue controller process.  A bug in prune.  A bad implementation.